### ISSUE TITLE
Cannot build on Debian 8.2 stable.

### ISSUE THREAD

**Issue by ik9999** [Eloston/ungoogled-chromium#2]
> [7588/18256] CXX obj/media/base/media.bitstream_buffer.o
warning: unknown warning option '-Wno-inconsistent-missing-override' [-Wunknown-warning-option]
warning: unknown warning option '-Wno-shift-negative-value'; did you mean '-Wno-shift-sign-overflow'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-bitfield-width' [-Wunknown-warning-option]
3 warnings generated.

>[7589/18256] CXX obj/media/base/media.audio_video_metadata_extractor.o
FAILED: /usr/bin/clang++ -MMD -MF obj/media/base/media.audio_video_metadata_extractor.o.d -DV8_DEPRECATION_WARNINGS -DCLD_VERSION=2 -D_FILE_OFFSET_BITS=64 -DNO_TCMALLOC -DDISABLE_NACL -DCHROMIUM_BUILD -DCR_CLANG_REVISION=247874-1 -DUI_COMPOSITOR_IMAGE_TRANSPORT -DUSE_AURA=1 -DUSE_ASH=1 -DUSE_PANGO=1 -DUSE_CAIRO=1 -DUSE_DEFAULT_RENDER_THEME=1 -DUSE_LIBJPEG_TURBO=1 -DUSE_X11=1 -DUSE_CLIPBOARD_AURAX11=1 -DENABLE_ONE_CLICK_SIGNIN -DENABLE_MEDIA_ROUTER=1 -DUSE_PROPRIETARY_CODECS -DENABLE_MPEG2TS_STREAM_PARSER -DENABLE_PEPPER_CDMS -DENABLE_CONFIGURATION_POLICY -DENABLE_NOTIFICATIONS -DENABLE_HIDPI=1 -DENABLE_TOPCHROME_MD=1 -DUSE_UDEV -DDONT_EMBED_BUILD_METADATA -DTRACING_IS_OFFICIAL_BUILD=1 -DENABLE_TASK_MANAGER=1 -DENABLE_EXTENSIONS=1 -DENABLE_PDF=1 -DENABLE_PLUGINS=1 -DENABLE_SESSION_SERVICE=1 -DENABLE_THEMES=1 -DENABLE_AUTOFILL_DIALOG=1 -DENABLE_BACKGROUND=1 -DENABLE_PRINTING=1 -DENABLE_BASIC_PRINTING=1 -DENABLE_PRINT_PREVIEW=1 -DENABLE_SPELLCHECK=1 -DENABLE_CAPTIVE_PORTAL_DETECTION=1 -DENABLE_APP_LIST=1 -DENABLE_SETTINGS_APP=1 -DENABLE_SUPERVISED_USERS=1 -DENABLE_MDNS=1 -DENABLE_SERVICE_DISCOVERY=1 -DV8_USE_EXTERNAL_STARTUP_DATA -DMEDIA_IMPLEMENTATION -DUSE_NEON -DUSE_ALSA -DUSE_PULSEAUDIO -DDLOPEN_PULSEAUDIO -DENABLE_HEVC_DEMUXING -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DU_STATIC_IMPLEMENTATION -DSK_SUPPORT_GPU=1 -DSK_IGNORE_LINEONLY_AA_CONVEX_PATH_OPTS -DSK_IGNORE_GL_TEXTURE_TARGET -D__STDC_CONSTANT_MACROS -DUSE_SYSTEM_FFMPEG -DUSE_LIBPCI=1 -DUSE_OPENSSL=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -D__STDC_FORMAT_MACROS -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -Igen -I../.. -Igen/pulse -I../../third_party/khronos -I../../gpu -I../../skia/config -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/skia/include/core -I../../third_party/skia/include/effects -I../../third_party/skia/include/pdf -I../../third_party/skia/include/gpu -I../../third_party/skia/include/lazy -I../../third_party/skia/include/pathops -I../../third_party/skia/include/pipe -I../../third_party/skia/include/ports -I../../third_party/skia/include/utils -I../../skia/ext -I../../third_party/libyuv/include -I../../third_party/libyuv -I../../third_party/opus/src/include -I../../third_party/libvpx_new/source/libvpx -fstack-protector --param=ssp-buffer-size=4  -pthread -fno-strict-aliasing -Wall -Wno-unused-parameter -Wno-missing-field-initializers -fvisibility=hidden -pipe -fPIC -Wno-reserved-user-defined-literal -fcolor-diagnostics -Wheader-hygiene -Wno-char-subscripts -Wno-unneeded-internal-declaration -Wno-covered-switch-default -Wstring-conversion -Wno-c++11-narrowing -Wno-deprecated-register -Wno-inconsistent-missing-override -Wno-shift-negative-value -Wno-bitfield-width -D_REENTRANT -pthread -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/nss -I/usr/include/nspr -Wno-header-guard -DCHROMIUM_OMIT_AV_PKT_DATA_MATROSKA_BLOCKADDITIONAL=1 -DCHROMIUM_NO_AVFRAME_CHANNELS=1 -m64 -march=x86-64 -O2 -fno-ident -fdata-sections -ffunction-sections -funwind-tables -D_FORTIFY_SOURCE=2 -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -Wsign-compare -std=gnu++11  -c ../../media/base/audio_video_metadata_extractor.cc -o obj/media/base/media.audio_video_metadata_extractor.o

>warning: unknown warning option '-Wno-inconsistent-missing-override' [-Wunknown-warning-option]

>warning: unknown warning option '-Wno-shift-negative-value'; did you mean '-Wno-shift-sign-overflow'? [-Wunknown-warning-option]

>warning: unknown warning option '-Wno-bitfield-width' [-Wunknown-warning-option]
../../media/base/audio_video_metadata_extractor.cc:116:17: error: use of undeclared identifier 'avcodec_get_name'; did you mean 'avcodec_get_type'?
    info.type = avcodec_get_name(stream->codec->codec_id);
                ^~~~~~~~~~~~~~~~
                avcodec_get_type
/usr/include/libavcodec/avcodec.h:4408:18: note: 'avcodec_get_type' declared here
enum AVMediaType avcodec_get_type(enum AVCodecID codec_id);
                 ^
../../media/base/audio_video_metadata_extractor.cc:93:32: warning: comparison of integers of different signs: 'int64_t' (aka 'long') and 'unsigned long' [-Wsign-compare]
  if (format_context->duration != AV_NOPTS_VALUE)
      ~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
>4 warnings and 1 error generated.

.....

>[7589/18256] CXX obj/media/base/media.container_names.o
warning: unknown warning option '-Wno-inconsistent-missing-override' [-Wunknown-warning-option]
warning: unknown warning option '-Wno-shift-negative-value'; did you mean '-Wno-shift-sign-overflow'? [-Wunknown-warning-option]
warning: unknown warning option '-Wno-bitfield-width' [-Wunknown-warning-option]
3 warnings generated.
ninja: build stopped: subcommand failed.
debian/rules:134: recipe for target 'override_dh_auto_build-arch' failed
make[1]: *** [override_dh_auto_build-arch] Error 1
make[1]: Leaving directory '/home/eq/Programs/ungoogled-chromium/build-sandbox'
debian/rules:127: recipe for target 'binary' failed
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2


Here is my log. Do you have any ideas on how to fix it?
Thank you.

**Comment by Eloston** [comment]
Debian 8 uses libav, but Debian 9 and later now use ffmpeg; there may be some API differences between them that's causing that error.

Fortunately, ffmpeg has been backported in `jessie-backports`. Try installing the `libavutil-dev`, `libavcodec-dev`, and `libavformat-dev` packages from there.

Then, if you haven't touched the building sandbox since that build error, you can run `dpkg-buildpackage -b -uc -nc` in `build-sandbox` to resume building.

**Comment by ik9999** [comment]
Thank you for your response.
I managed to finish making, but got this error:
>make[1]: Leaving directory '/home/eq/Programs/ungoogled-chromium/build-sandbox'
   debian/rules override_dh_strip
make[1]: Entering directory '/home/eq/Programs/ungoogled-chromium/build-sandbox'
> this line can be removed once stretch is released
dh_strip --ddeb-migration='chromium-dbg (<< 47.0.2526.80-4~)'
dh_strip: unknown option; aborting
debian/rules:166: recipe for target 'override_dh_strip' failed
make[1]: *** [override_dh_strip] Error 25
make[1]: Leaving directory '/home/eq/Programs/ungoogled-chromium/build-sandbox'
debian/rules:127: recipe for target 'binary' failed
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2

Is it the problem in chromium itself?

**Comment by Eloston** [comment]
It looks like debhelper in Debian 8 is too old, so the option `ddeb-migration` is not supported.

Just remove the following lines from `build-sandbox/debian/rules` and re-run the build command I provided earlier:
```
 override_dh_strip:
	# this line can be removed once stretch is released
	dh_strip --ddeb-migration='chromium-dbg (<< 47.0.2526.80-4~)'
```

**Comment by ik9999** [comment]
Managed to build. Thank you very much for your support and for this project.
Also, for my dpkg version(1.17.26) i needed to edit download_source.sh(i've removed space after -l)
>-    DOWNLOAD_VERSION=$(dpkg-parsechangelog -l $DEBIAN_CHANGELOG -S Version | sed s/-.*//);
>+    DOWNLOAD_VERSION=$(dpkg-parsechangelog -l$DEBIAN_CHANGELOG -S Version | sed s/-.*//);

**Comment by Eloston** [comment]
That's interesting. They must have changed the syntax in the new `dpkg-parsechangelog` since the manpage shows there should be a space (for version 1.18.4).

Anyways, I'm glad I could help.

**Comment by kirill9000** [comment]
Hello. I've just run into same three issues (space after -l in dpkg-parsechangelog arguments, old libav and override_dh_strip) in while building on Debian Stable. Could you please fix the scripts so they would check distro version and work on stable without manual modifications? Oh, and mentioning jessie-backports in README would be nice too. Thanks!

**Comment by ik9999** [comment]
About jessie-backports, i added this to my sources.list:
>deb http://ftp.se.debian.org/debian/ jessie-backports main
>deb-src http://ftp.se.debian.org/debian/ jessie-backports main
>deb http://http.debian.net/debian jessie-backports main
>deb-src http://http.debian.net/debian jessie-backports main
Then did:
>sudo apt-get install libavutil-dev/jessie-backports libavcodec-dev/jessie-backports libavformat-dev/jessie-backports

**Comment by kirill9000** [comment]
@ik9999 Yeah, I figured that and successfully built Chromium with your and @Eloston's instructions. Just thought that having this information in README would help since a lot of people use Stable.

**Comment by Eloston** [comment]
Changes implemented in 7f45ba754b77b89838d96b210fabbb379106fec2

**Comment by kirill9000** [comment]
Works like a charm now, thank you.
